consensus/types/src/blob_sidecar.rs

```diff
@@ -149,6 +149,37 @@ impl<E: EthSpec> BlobSidecar<E> {
         })
     }
 
+    pub fn new_efficiently(
```

Comment:
Maybe a better name `new_with_existing_proof`

Reply:
Updated in https://github.com/sigp/lighthouse/pull/6268/commits/dbf4bb8bfdbb3ca37d82b948958ef48670aa0882